app/services/translation/sarvam_client.py

```
@@ -0,0 +1,68 @@
+                    data = response.json()
+                    # Assuming response format: {"translated_text": "..."}
+                    # Documentation may vary, typically it's translated_text
+                    return data.get("translated_text", text)
+                elif response.status_code in [429, 503]:
+                    wait = 2 ** attempt * 5
+                    logger.warning(f"Sarvam AI rate limited (status {response.status_code}), attempt {attempt+1}, waiting {wait}s...")
+                    time.sleep(wait)
+                else:
+                    logger.error(f"Sarvam API error {response.status_code}: {response.text}")
+                    raise TranslatorError(f"Sarvam API error {response.status_code}: {response.text}")
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)